test/13_arraycommon.jl

```diff
@@ -2260,4 +2260,22 @@ end
         Vector{CategoricalVector{<:Any, <:Integer, <:Any, <:Any, Union{}}}
 end
 
+@testset "levels with skipmissing argument" begin
+    for x in (categorical(["a", "b", "a"], levels=["b", "c", "a"]),
+              view(categorical(["c", "b", "a"], levels=["b", "c", "a"]), 2:3))
+        @test @inferred(levels(x)) == ["b", "c", "a"]
+        @test @inferred(levels(x, skipmissing=true)) == ["b", "c", "a"]
+        @test @inferred(levels(x, skipmissing=false)) == ["b", "c", "a"]
+    end
+
+    for x in (categorical(Union{String, Missing}["a", "b", "a"], levels=["b", "c", "a"]),
+              view(categorical(Union{String, Missing}["c", "b", "a"], levels=["b", "c", "a"]), 2:3))
+        @test @inferred(levels(x)) == ["b", "c", "a"]
+        @test levels(x, skipmissing=true) == ["b", "c", "a"]
+        @test levels(x, skipmissing=true) isa Vector{String}
+        @test levels(x, skipmissing=false) == ["b", "c", "a"]
+        @test levels(x, skipmissing=false) isa Vector{Union{String, Missing}}
```

Comment:
I couldn't find a way to make this inferrable, even by wrapping this in a function. Not sure why. It's not terrible since the return type is inferred as `Union{Vector{Union{String, Missing}}, Vector{String}}`. But still...